### PR TITLE
Distribution initial test

### PR DESCRIPTION
**Pull request by kmilstein**
Created a interface for EmailService and changed the name of it to
EmailServiceImpl. Changed setEmailServiceMethod in EmailScheduler to
reflect name change. Created DistributionController and DistributionService
classes. Created tests for EmailService and DistributionController.

### CODE REVIEW

File: SurveyService/src/main/java/com/revature/controller/DistributionController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.Set;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller takes in a batch id, or a batch id with a set of emails, then
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/controller/DistributionController.java]: Place validation description in Service where it belongs.  Have this specifically explain purpose of rest endpoints.

File: SurveyService/src/main/java/com/revature/controller/DistributionController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	 * 
+	 * @param batchId represents a batch identifier
+	 * @return List of incorrectly formatted emails in the database if any
+	 */
+	@PostMapping("/distribute/{batchId}")
+	private ResponseEntity<List<String>> sendEmailsByBatchId(@PathVariable int batchId) {
+
+		return null;
+	}
+
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/controller/DistributionController.java]: not documented

File: SurveyService/src/main/java/com/revature/service/DistributionServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.revature.service;
+
+import java.io.File;
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/service/DistributionServiceImpl.java:14]: no documentation

File: SurveyService/src/main/java/com/revature/service/EmailServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		    }
+		});  
+	}
+
+	/**Sends an email to a particular email address
+	 * @param msg contents of the email 
+	 * @param destination email address
+	 * @throws AddressException exception thrown when a wrongly formatted address is encountered.
+	 * @throws MessagingException base class for all exceptions thrown by the Messaging classes
+	 */
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/service/EmailServiceImpl.java:77]: update method name to send email singular

File: SurveyService/src/main/java/com/revature/service/EmailService.java [hunk 5]
```diff
@@ -1,97 +1,10 @@
...
-	 */
-	@Value("${survey_service.email_service.username}")
-	private String emailServiceUsername;
 
-	/**
-	 * The password of the email account
-	 */
-	@Value("${survey_service.email_service.password}")
-	private String emailServicePassword;
+public interface EmailService {
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/service/EmailService.java:7]: change to send email singular

File: SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	void distributionControllerSendEmailsByBatchId_withoutError() throws Exception {
+
+		final List<String> invalidEmails = new ArrayList<>(Arrays.asList());
+
+		int validBatchId = 2010;
+
+		Mockito.when(service.sendEmailsByBatchId(validBatchId)).thenReturn(invalidEmails);
+
+		RequestBuilder request = MockMvcRequestBuilders.post("/distribute/" + validBatchId);
+		MvcResult result = mockMvc.perform(request).andReturn();
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:58]: add verification service is called

File: SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+	void distributionControllerSendEmailsByBatchId_withInvalidEmail() throws Exception {
+
+		final List<String> invalidEmails = new ArrayList<>(Arrays.asList("acacia.hollidayrevature.net"));
+
+		int validBatchId = 2010;
+
+		Mockito.when(service.sendEmailsByBatchId(validBatchId)).thenReturn(invalidEmails);
+
+		RequestBuilder request = MockMvcRequestBuilders.post("/distribute/" + validBatchId);
+		MvcResult result = mockMvc.perform(request).andReturn();
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java]: check status code as well

File: SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+import com.revature.util.InvalidBatchIdException;
+
+/*
+ * This tests the sendEmailsByBatchId method in the Distribution controller. If there are no errors, then a it will perform a 
+ * post request and not return any emails. If an invalid batchId is provided, an exception will be thrown and a status of
+ * Bad Request, if a batchId returns incorrectly formatted emails from the service the controller returns the list of invalid emails. 
+ */
+@SpringBootTest
+@AutoConfigureMockMvc
+@WebMvcTest
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:35]: test for improperly formatted .csv

File: SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+		int validBatchId = 2010;
+
+		Mockito.when(service.sendEmailsByBatchId(validBatchId)).thenReturn(invalidEmails);
+
+		RequestBuilder request = MockMvcRequestBuilders.post("/distribute/" + validBatchId);
+		MvcResult result = mockMvc.perform(request).andReturn();
+		assertEquals("", result.getResponse().getContentAsString());
+
+	}
+
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:65]: comment test

**Review by jurczakn — CHANGES_REQUESTED** [review]
update, and notify when fixed

File: SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java (hunk 8 above)
Comment (kmilstein) [reply SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:35]: Test has been added

File: SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java (hunk 9 above)
Comment (kmilstein) [reply SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:65]: Commends have been added

**Comment by kmilstein** [comment]
Requested changes have been made.

File: SurveyService/src/main/java/com/revature/controller/DistributionController.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+	/**
+	 * The method will send an email to associates that are in the database for the
+	 * batchId provided.
+	 * 
+	 * @param batchId represents a batch identifier
+	 * @return List of incorrectly formatted emails in the database if any
+	 */
+	@PostMapping("/distribute/{batchId}")
+	private ResponseEntity<List<String>> sendEmailsByBatchId(@PathVariable int batchId) {
+
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/controller/DistributionController.java:31]: add TODO comments

File: SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+@SpringBootTest
+@AutoConfigureMockMvc
+@WebMvcTest
+class DistributionControllerTest {
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@MockBean
+	private DistributionService service;
```
Comment (jurczakn) [inline SurveyService/src/test/java/com/revature/controller/DistributionControllerTest.java:42]: improve name

File: SurveyService/src/main/java/com/revature/service/DistributionServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+	public List<String> sendEmailsByBatchId(int batchId) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	/**
+	 * TODO: Document after creating tests
+	 */
+	@Override
+	public List<String> sendEmailsByBatchIdAndCSV(int batchId, MultipartFile csv) {
```
Comment (jurczakn) [inline SurveyService/src/main/java/com/revature/service/DistributionServiceImpl.java:30]: have throw more specific exceptions

**Review by jurczakn — CHANGES_REQUESTED** [review]

**Review by jurczakn — APPROVED** [review]

**Pull request merged**